presto-main/src/main/java/com/facebook/presto/operator/aggregation/ApproximateLongPercentileAggregations.java

```diff
@@ -32,19 +32,37 @@
 @AggregationFunction("approx_percentile")
 public final class ApproximateLongPercentileAggregations
 {
+    static final double DEFAULT_ACCURACY = 0.01;
+
     private ApproximateLongPercentileAggregations() {}
 
     @InputFunction
-    public static void input(@AggregationState DigestAndPercentileState state, @SqlType(StandardTypes.BIGINT) long value, @SqlType(StandardTypes.DOUBLE) double percentile)
+    public static void input(
```

Comment:
Could all these input functions just call weightedInput with a weight of 1.0? It looks like in QuantileDigest add without a weight is just add with a weight of 1. Then a lot of duplicated code could be removed.

Reply:
We could!  We'd end up checking the weight every row (checking that `weight > 0`).  Do we have a feel for the performance implications?  It's worth noting that in the case of accuracy, the check is only done once (when the digest is created).

Reply:
I suppose we could have a private helper method that didn't check, and everything delegate to that.

Reply:
Sounds like you have a plan :-)